File: examples/java/S3ListBuckets.java
// Lista todos los buckets S3 existentes en Floci.
// Requiere: software.amazon.awssdk:s3 (AWS SDK for Java v2).
import java.net.URI;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.s3.S3Client;

public class S3ListBuckets {
  public static void main(String[] args) {
    S3Client s3 = S3Client.builder()
        .endpointOverride(URI.create("http://localhost:4566"))
        .region(Region.US_EAST_1)
        .credentialsProvider(
            StaticCredentialsProvider.create(AwsBasicCredentials.create("test", "test")))
        .forcePathStyle(true)
        .build();

    try {
      var buckets = s3.listBuckets().buckets();
      if (buckets.isEmpty()) {
        System.out.println("No hay buckets todavía. Crea uno con S3CreateBucket.");
        return;
      }
      System.out.println("Buckets:");
      buckets.forEach(b -> System.out.println("  - " + b.name() + " (creado " + b.creationDate() + ")"));
    } finally {
      s3.close();
    }
  }
}

File: examples/node/s3-list-buckets.js
// Lista todos los buckets S3 existentes en Floci con AWS SDK v3.
// Uso (desde examples/node): npm install && node s3-list-buckets.js
const { S3Client, ListBucketsCommand } = require('@aws-sdk/client-s3');

const s3 = new S3Client({
  endpoint: 'http://localhost:4566',
  region: 'us-east-1',
  credentials: { accessKeyId: 'test', secretAccessKey: 'test' },
  // Floci local resuelve mejor las URLs como /bucket/objeto que como subdominios.
  forcePathStyle: true,
});

async function main() {
  const result = await s3.send(new ListBucketsCommand({}));
  const buckets = result.Buckets ?? [];
  if (!buckets.length) {
    console.log('No hay buckets todavía. Crea uno con s3-create-bucket.js');
    return;
  }
  console.log('Buckets:');
  buckets.forEach((bucket) => {
    const creationDate = bucket.CreationDate?.toISOString() ?? 'fecha no disponible';
    console.log(`  - ${bucket.Name} (creado ${creationDate})`);
  });
}

main().catch((error) => {
  console.error('Error listando buckets:', error.message);
  process.exit(1);
});
